aiohttp/client.py

```diff
@@ -610,7 +610,7 @@ async def _request(
                         is_same_host_https_redirect = (
                             url.host == parsed_url.host
                             and parsed_url.scheme == "https"
-                            and url.scheme == "http"
+                            and url.scheme in ("http", "https")
                         )
 
                         if (
```

Comment:
@greshilov @webknjaz Do you think this could maybe just be reduced to:
```
if url.host != parsed_url.host or parsed_url.scheme != "https":
```

i.e. Keep the Authorization header if it's the same host and the redirect is to https. Do we actually need to care whether it's the same port or not?

If we do care about the port, then maybe this needs to be fixed in yarl. We get 2 different origins for 'https://foo.com', and 'https://foo.com:443', despite both reporting the same port.

Reply:
1. the original bug mentioned a URL with port vs one without port, so we do care
2. Maybe it should be fixed in yarl, but that will probably constitute a breaking change, and will take time to pass through
3. I do agree on simplifying the statement

Reply:
> the original bug mentioned a URL with port vs one without port, so we do care

Maybe you misread that. If we don't care, as per my suggested code, then it works regardless of the port number. If we do care, then we should make this code _more_ restrictive, not less, and the problem can only be solved in yarl at that point.

Reply:
> Maybe it should be fixed in yarl, but that will probably constitute a breaking change, and will take time to pass through

@mjpieters Do you think this change in behaviour would be acceptable in a minor release?

i.e. We want to make this test pass: `URL("https://foo.com").origin() == URL("https://foo.com:443").origin()`

Reply:
> separating protection spaces by using a different host name (or port number) for each party.
https://www.rfc-editor.org/rfc/rfc9110.html#section-17.16.3-2

This suggests to me that we should care about the port number. In which case, it would be best to fix in yarl.
I guess http->https will (almost) always change the port number, so we probably can't be more restrictive in that specific case.